server/BootstrapServer.java

```diff
@@ -177,23 +203,39 @@ public static void main(String args[])
                     String ip = st.nextToken();
                     int port = Integer.parseInt(st.nextToken());
                     String username = st.nextToken();
+                    boolean found = false;
                     for (int i=0; i<nodes.size(); i++) {
                         if (nodes.get(i).getPort() == port) {
```

Comment:
@shafraz17 @PasinduUkwatta Are we checking only the port for the node connectivity? this seems incorrect,
two nodes might communicate using the same port, only the `ip:port` combination should be unique for a node.

OR

Am I missing something?

Reply:
This was the provided server code. This is related to the UNREG command right. according to the code, it only checks for the port right when removing a node

Reply:
if this part is provided then we dont want to worry, however we need to ensure that this does not affect impl.